packages/fxa-auth-server/lib/payments/paypal.ts

```diff
@@ -346,6 +346,36 @@ export class PayPalHelper {
     }));
   }
 
+  /**
+   * Removes Paypal billing agreements on a customer if they paid with
+   * Paypal but no longer have an active/past_due subscription.
+   */
+  async conditionallyRemoveBillingAgreement(
+    customer: Stripe.Customer
+  ): Promise<boolean> {
+    const billingAgreementId = this.stripeHelper.getCustomerPaypalAgreement(
+      customer
+    );
+    if (!billingAgreementId) {
+      return false;
+    }
+    const paypalSubscription = customer.subscriptions?.data.find(
+      (sub) =>
+        ['active', 'past_due'].includes(sub.status) &&
+        sub.collection_method === 'send_invoice'
+    );
+    if (paypalSubscription) {
+      return false;
+    }
+    await this.cancelBillingAgreement(billingAgreementId);
+    await this.stripeHelper.removeCustomerPaypalAgreement(
+      customer.metadata.userid,
+      customer,
+      billingAgreementId
+    );
+    return true;
+  }
```

Comment:
Why is this function returning a bool?  That return value is ultimately ignored in `handleWebhookEvent`.

Reply:
I do see that you are using it in tests and it does make that easier.

Reply:
While it doesn't need to return anything, it does help as another safe-guard on testing to see the branch went the right way.